### DISCUSSION TITLE
Prevent Line-breaks in ternary operators in Beautifier

### DISCUSSION THREAD

**Discussion by Ivan-devprat** [phcode-dev/phoenix#1924] [Q&A] [answered]
Hi folks !
When I beautify with Cmd+B in my PHP code, it tends to add line break in my ternary operations.
For example, the following line :
```php 
echo (array_key_exists('url',$btn))?$btn['url']:'#';
```

Will become : 
```php
echo array_key_exists("url", $btn)
            ? $btn["url"]
            : "#";
```

Maybe it is some kind of convention, but I do not want it to do so. I couldn't find any options in defaultPreferences to add.

Do you know if that's possible or should I just use a plugin for beautifying specialised in PHP ?
Thank you !

**Comment by abose** [comment]
Hello there! We use prettier for that and any options supported by prettier for php will work in phoenix too. see https://docs.phcode.dev/docs/Features/beautify-code

If it doesn't work for you, can you suggest a PHP beautifier you use, if you have used any. We could investigate of we cam add that feature.

**Reply by Ivan-devprat** [reply]
Nevermind what I said before.
I found the options but unfortunately, its doesn't work. Guess I'll find a way somehow !
Here what I used
```json
{
    "beautify.options": {
        "experimentalTernaries": false
    },
}
```

Thanks for helping me !

**Comment by Ivan-devprat** [comment, marked as answer]
So, I think I found the problem.
Simply put, Prettier doesn't work well with mixed php/html files. Hence, this issue that appear when I do inline PHP.

To quote Prettier:

> We're considering the plugin to be stable when pure PHP files are formatted. Formatting of files that contain mixed PHP and HTML is still considered unstable - please see [open issues with the tag "inline"](https://github.com/prettier/plugin-php/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Ainline) for details.

Beautify should then be used with care while editing mixed HTML/PHP files.

Thanks to @abose for helping with this !
